SCSH_3C_iOS/SCSH_3C_iOS/Modules/Home/View/HomeViewController.swift

```diff
@@ -0,0 +1,196 @@
+            bodyCollectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            bodyCollectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            bodyCollectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+}
+
+extension HomeViewController: UICollectionViewDelegateFlowLayout {
+    ///
+    /// The layout of Header & Body.
+    ///
+    /// Header
+    ///         []  []  []
+    /// Body
+    ///         [][][][][]
+    ///
+    /// When the Body swips `5` blocks, the Header needs to swipe `2` blocks.
+    ///
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard scrollView == bodyCollectionView else { return }
+        headerCollectionView.contentOffset.x = scrollView.contentOffset.x * (2 / 5)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard collectionView == bodyCollectionView else { return }
+        print("BodyCollectionView selected cell at \(indexPath.item)")
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        infinitScrollItems
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        
+        switch collectionView {
+        case headerCollectionView:
+            return CGSize(width: collectionView.bounds.width / 5, height: collectionView.bounds.height / 2)
+        case bodyCollectionView:
+            return CGSize(width: collectionView.bounds.width, height: collectionView.bounds.height)
+        default:
+            return .zero
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+       
+        switch collectionView {
+        case headerCollectionView:
+            return collectionView.bounds.width / 5
+        case bodyCollectionView:
+            return 0
+        default:
+            return 0
+        }
+    }
+}
+
+extension HomeViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        switch collectionView {
+        case headerCollectionView:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "HeaderCollectionViewCell", for: indexPath) as! HeaderCollectionViewCell
+            
+            // get the actual index for subjects array
+            var index = indexPath.item % viewModel.subjects.count
+            
+            // Since one page contains three header cells, the primary cell should be centered on the page.
+            if index == 0 {
+                index = viewModel.subjects.count - 1
+            } else {
+                index -= 1
+            }
```

Comment:
nit: How about using ternary operator?

```swift
index = (index == 0)
    ? viewModel.subjects.count - 1
    : index - 1
```

Reply:
I rarely used the ternary operator unless the logic was simple. 

In this case, I prefer using `IF-ELSE` control flow.